Detectors/ITSMFT/ITS/macros/test/CheckTopologies.C

```diff
@@ -99,107 +95,104 @@ void CheckTopologies(std::string clusfile = "o2clus_its.root", std::string hitfi
+          hitTree->GetEntry(im);
+          auto& mc2hit = mc2hitVec[im];
+          const auto* hitArray = hitVecPool[im];
+          for (int ih = hitArray->size(); ih--;) {
+            const auto& hit = (*hitArray)[ih];
+            uint64_t key = (uint64_t(hit.GetTrackID()) << 32) + hit.GetDetectorID();
+            mc2hit.emplace(key, ih);
           }
         }
-      } // << cache MC events contributing to this ROF
-
-      for (int icl = 0; icl < rofRec.getNEntries(); icl++) {
-        int clEntry = rofRec.getFirstEntry() + icl; // entry of icl-th cluster of this ROF in the vector of clusters
-        // do we read MC data?
-        const auto& cluster = (*clusArr)[clEntry];
-        ClusterTopology topology;
-        o2::itsmft::ClusterPattern pattern(pattIdx);
-        topology.setPattern(pattern);
-        //output_check << "iEv: " << ievC << " / " << nevCl << " iCl: " << clEntry << " / " <<  clusArr->size() << std::endl;
-        // output_check << topology << std::endl;
-
-        const auto locC = o2::itsmft::TopologyDictionary::getClusterCoordinates(cluster, pattern);
-        float dx = BuildTopologyDictionary::IgnoreVal * 2, dz = BuildTopologyDictionary::IgnoreVal * 2; // to not use unassigned dx,dy
-        if (clusLabArr) {
-          const auto& lab = (clusLabArr->getLabels(clEntry))[0];
-          auto srcID = lab.getSourceID();
-          if (lab.isValid() && srcID != QEDSourceID) { // use MC truth info only for non-QED and non-noise clusters
-            int trID = lab.getTrackID();
-            const auto& mc2hit = mc2hitVec[lab.getEventID()];
-            const auto* hitArray = hitVecPool[lab.getEventID()];
-            Int_t chipID = cluster.getSensorID();
-            uint64_t key = (uint64_t(trID) << 32) + chipID;
-            auto hitEntry = mc2hit.find(key);
-            if (hitEntry != mc2hit.end()) {
-              const auto& hit = (*hitArray)[hitEntry->second];
-              auto locH = gman->getMatrixL2G(chipID) ^ (hit.GetPos()); // inverse conversion from global to local
-              auto locHsta = gman->getMatrixL2G(chipID) ^ (hit.GetPosStart());
-              locH.SetXYZ(0.5 * (locH.X() + locHsta.X()), 0.5 * (locH.Y() + locHsta.Y()), 0.5 * (locH.Z() + locHsta.Z()));
-              const auto locC = o2::itsmft::TopologyDictionary::getClusterCoordinates(cluster, pattern);
-              dx = locH.X() - locC.X();
-              dz = locH.Z() - locC.Z();
-            } else {
-              printf("Failed to find MC hit entry for Tr:%d chipID:%d\n", trID, chipID);
-              continue;
-            }
-            signalDictionary.accountTopology(topology, dx, dz);
+      }
+    } // << cache MC events contributing to this ROF
+
+    for (int icl = 0; icl < rofRec.getNEntries(); icl++) {
+      int clEntry = rofRec.getFirstEntry() + icl; // entry of icl-th cluster of this ROF in the vector of clusters
+      // do we read MC data?
+
+      const auto& cluster = (*clusArr)[clEntry];
+
```

Comment:
The macro currently crashes if run over the clusters produced with a dictionary. Could you please a check for 
``cluster.getPatternID()  == CompCluster::InvalidPatternID`` and print a meaningful warning and quit if the check is not passed?
Even smarter would be to let the macro digest any data (i.e. to rebuild the dictionary): in case it is not an "invalid pattern" or "group pattern", extract its topology from the existing dictionary, otherwise get it from the ``pattIdx`` vector.
But this can be done later.
Could you put here a protection